### ISSUE TITLE
error[E0015]: cannot call non-const fn `Identity::new` in constant functions

### ISSUE THREAD

**Issue by RemiBardon** [tower-rs/tower#784]
I just added `tower` to my project, so I naturally chose version `v0.5.0` (just-released). Unfortunately it doesn't build:

```log
   Compiling tower v0.5.0
error[E0015]: cannot call non-const fn `Identity::new` in constant functions
   --> ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tower-0.5.0/src/builder/mod.rs:120:20
    |
120 |             layer: Identity::new(),
    |                    ^^^^^^^^^^^^^^^
    |
    = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
```

`v0.4.13` builds successfully so I switched to it for the time being.

I'm running:

```txt
rustup 1.27.1 (54dd3d00f 2024-04-24)
rustc 1.82.0-nightly (91376f416 2024-08-12)
```

**Comment by seanmonstar** [comment]
I see that tower-layer also released a new version making that function `const`. So you'd need to make sure to `cargo update` both tower and tower-layer to get them.

Within tower, we could bump the minimal version to make sure it's always picked up...

**Comment by RemiBardon** [comment]
Ok so I added the two dependencies at the worse timing then 😅 I added `tower-http` a few hours ago, before `tower v0.5.0` was released

**Comment by RemiBardon** [comment]
That was it, sorry for bothering

**Comment by agourlay** [comment]
Ran into the same issue, decided to implement the proper fix in https://github.com/tower-rs/tower/pull/787

**Comment by praseodym** [comment]
We hit this issue to on a Dependabot upgrade from Tower 0.4.13 to 0.5.0. Maybe a 0.5.1 release with the fix from #787 could prevent more people from running into this?
